src/interface/state_descriptor.cpp

```diff
@@ -271,6 +271,20 @@ bool StateDescriptor::AddFieldImpl(const VarID &vid, const Metadata &m_in,
   if (FieldPresent(vid.label()) || SparseBaseNamePresent(vid.label())) {
     return false; // this field has already been added
   } else {
+    if (m.IsSet(Metadata::WithFluxes) && m.GetFluxName() == "") {
+      std::vector<MetadataFlag> mFlags = {Metadata::OneCopy, Metadata::Flux};
+      if (m.IsSet(Metadata::Sparse)) mFlags.push_back(Metadata::Sparse);
+      if (m.IsSet(Metadata::Cell))
+        mFlags.push_back(Metadata::Face);
+      else if (m.IsSet(Metadata::Face))
+        mFlags.push_back(Metadata::Edge);
+      else if (m.IsSet(Metadata::Edge))
+        mFlags.push_back(Metadata::Node);
+      Metadata mf(mFlags, m.Shape());
+      auto fId = VarID{"bnd_flux::" + vid.base_name, vid.sparse_id};
+      AddFieldImpl(fId, mf, control_vid);
+      m.SetFluxName(fId.label());
+    }
```

Comment:
I'm not sure why, but I find the location of this logic in `StateDescriptor` a little distasteful. But I don't know of a better place for it.

Reply:
The only other place I could think is to add a static method `Metadata::MakeFluxFieldMetadata(const Metadata& in)`. But the calls to `AddFieldImpl` and `SetFluxName` would still probably need to be here.

Reply:
Yeah... That feels a little better to me, but I honestly can't express why. I think this is fine as is for now.